src/configuration.cpp

```diff
@@ -161,74 +160,154 @@ chimera::CompiledConfiguration::CompiledConfiguration(
         namespaces_.insert(ns);
     }
 
-    // Resolve namespace configuration entries within provided AST.
-    for(const auto &it : configNode["namespaces"])
+    // Get a reference to the configuration YAML root node.
+    // If it is not available, then skip the remaining parsing.
+    if (!configNode_)
+        return;
+
+    // Parse 'namespaces' section of configuration YAML if it exists.
+    const YAML::Node &namespacesNode = configNode_["namespaces"];
+    if (namespacesNode)
     {
-        std::string ns_str = it.first.as<std::string>();
-        auto ns = chimera::util::resolveNamespace(ci, ns_str);
-        if (ns)
+        // Check that 'namespaces' node in configuration YAML is a map.
+        if (!namespacesNode.IsMap())
         {
-            declarations_[ns] = it.second;
-            namespaces_.insert(ns);
+            std::cerr << "'namespaces' in configuration YAML must be a map."
+                      << std::endl;
+            exit(-2);
         }
-        else
+
+        // Resolve namespace configuration entries within provided AST.
+        for(const auto &it : namespacesNode)
         {
-            std::cerr << "Unable to resolve namespace: "
-                      << "'" << ns_str << "'." << std::endl;
-            exit(-1);
+            std::string ns_str = it.first.as<std::string>();
+            auto ns = chimera::util::resolveNamespace(ci, ns_str);
+            if (ns)
+            {
+                declarations_[ns] = it.second;
+                namespaces_.insert(ns);
+            }
+            else
+            {
+                std::cerr << "Unable to resolve namespace: "
+                          << "'" << ns_str << "'." << std::endl;
+                exit(-2);
+            }
         }
     }
 
-    // Resolve class/struct configuration entries within provided AST.
-    for(const auto &it : configNode["classes"])
+    // Parse 'classes' section of configuration YAML if it exists.
+    const YAML::Node &classesNode = configNode_["classes"];
```

Comment:
This is technically undefined behavior because `configNode` returns a `Node` by value. This issue is prevalent throughout Chimera - and a few of my other projects that use `yaml-cpp` 😅 - so it's best to make that change in a followup pull request.